sys-boot/plymouth/plymouth-9999.ebuild

```diff
@@ -1,92 +1,102 @@
-# Copyright 1999-2023 Gentoo Authors
+# Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI=8
+
+inherit meson readme.gentoo-r1 flag-o-matic
 
-inherit flag-o-matic
 SRC_URI="https://dev.gentoo.org/~aidecoe/distfiles/${CATEGORY}/${PN}/gentoo-logo.png"
 
 if [[ ${PV} == 9999 ]]; then
 	inherit git-r3
 	EGIT_REPO_URI="https://gitlab.freedesktop.org/plymouth/plymouth"
 else
 	SRC_URI="${SRC_URI} https://www.freedesktop.org/software/plymouth/releases/${P}.tar.xz"
-	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~loong ~ppc ~ppc64 ~riscv ~sparc ~x86"
 fi
 
-inherit autotools readme.gentoo-r1 systemd
-
 DESCRIPTION="Graphical boot animation (splash) and logger"
 HOMEPAGE="https://cgit.freedesktop.org/plymouth/"
 
-LICENSE="GPL-2"
+LICENSE="GPL-2+"
 SLOT="0"
-IUSE="debug +drm +gtk +pango selinux +split-usr static-libs +udev"
+IUSE="debug +drm +gtk +pango selinux freetype +split-usr +udev doc upstart-monitoring systemd"
 
-CDEPEND="
+BDEPEND="
+	app-text/docbook-xsl-stylesheets
+	>=dev-build/meson-0.62
+	dev-libs/libxslt
+	virtual/pkgconfig
+"
+
+DEPEND="
+	dev-libs/libevdev
 	>=media-libs/libpng-1.2.16:=
+	virtual/libc
+	x11-libs/libxkbcommon
+	x11-misc/xkeyboard-config
 	drm? ( x11-libs/libdrm )
+	elibc_musl? ( sys-libs/rpmatch-standalone )
+	freetype? ( media-libs/freetype:2 )
 	gtk? (
 		dev-libs/glib:2
-		>=x11-libs/gtk+-3.14:3
 		x11-libs/cairo
+		>=x11-libs/gtk+-3.14:3[X]
+	)
+	pango? (
+		x11-libs/cairo
+		>=x11-libs/pango-1.21
+	)
+	systemd? ( sys-apps/systemd )
+	udev? ( virtual/libudev )
+	upstart-monitoring? (
+		sys-apps/dbus
+		sys-libs/ncurses
 	)
-	pango? ( >=x11-libs/pango-1.21 )
-"
-DEPEND="${CDEPEND}
-	elibc_musl? ( sys-libs/rpmatch-standalone )
-	app-text/docbook-xsl-stylesheets
-	dev-libs/libxslt
-	virtual/pkgconfig
 "
-# Block due bug #383067
-RDEPEND="${CDEPEND}
+
+RDEPEND="${DEPEND}
+	>sys-kernel/dracut-0.37-r3
```

Comment:
Is dracut really always required in this version? If so, then the warning in postinst does not make sense any more.

Reply:
Good spot, the version requirement was there in previous ebuilds: `!<sys-kernel/dracut-0.37-r3`, however I'm **assuming** that dracut is not needed at all.
The warning was also from previous ebuilds, however, dracut should indeed probably be removed, which would then make the warning make sense.

Reply:
dracut is NOT a version requirement in previous ebuilds.

The ! dependency in the previous ebuilds means that *iff* you have dracut installed, and you try to install plymouth, you cannot have an *old* version of plymouth. This originates back to https://bugs.gentoo.org/512436

You could remove the ! dependency because there's no need to explicitly mark plymouth as "extremely incompatible with" version 0.37 of dracut, as that version is long gone from gentoo (the restriction dates back to 2014). But that would not be removing a "dependency", that would be removing a "restriction".